proposals/nnnn-video-streaming-state.md

```diff
@@ -0,0 +1,165 @@
+    <description>Projection app is shown in HMI</description>
+  </element>
+</enum>
+```
+
+#### Compatibility
+- New apps on old HUs: Old HUs do not support new projection type app. Since old HUs does not send the new videoStreamingState parameter, on mobile proxy, we can give default value as *streamable* to  the parameter and make sure app does not do streaming in `BACKGROUND` or `NONE`  to make new navigation apps work with old HUs.
+- Old apps on new HUs: the old navigation app does not recognize the new parameter, thus ignores it. Since there are navigation apps only (no old projection apps), the app use HMI level as an indication to stop video streaming as it current does. If a new non-media projection app is launched, the old Nav app moves to LIMITED and continues to stream its video. SDL shall send stop service control frame to the app.
+
+
+## Potential downsides
+On iOS devices, we still lack an effective way to bring a background Navigation/Projection app to foreground so that the app can restart/resume video streaming when the driver brings the app back to `FULL` in HMI.
+
+
+## Impact on existing code
+
+- RPC need get updated with new parameter. 
+- SDL core and mobile proxy lib need updates to support this new parameter. 
+- App's HMI level transition logic in SDL core need updates.
+
+
+## Alternatives considered
+
+- Instead of using a new enum data type `VideoStreamingState`, we can use just `Boolean` type and rename `videoStreamingState` to `isVideoSource` as the following.
+
+```xml
+<function name="OnHMIStatus" functionID="OnHMIStatusID" messagetype="notification">
+    ...
+    
+    <!-- new additions-->
+    <param name="isVideoSource" type="Boolean" mandatory="true">
+        <description>Indicates whehter an app can use video service or not.
+        If it is true, the mobile app can use video service.
+        If it is false, the mobile app must stop streaming video to SDL.        
+        </description>
+    </param>
+</function>
+```
+
+- Another alternative is that we can seperate one `STREAMABLE` state into two sub-states : `STREAMABLE_VISIBLE` and `STREAMABLE_NOT_VISIBLE`. However, this makes videoStreamingState coupled with HMI level. `VISIBLE` must be with `FULL` and `NOT_VISIBLE` must be with `LIMITED`.
+
+```xml
+<enum name="VideoStreamingState">
+    <description>Enumeration that describes possible states of video streaming. </description>
+    <element name="STREAMABLE_VISIBLE" />
+    <element name="STREAMABLE_NOT_VISIBLE" />
+    <element name="NOT_STREAMABLE" />
+</enum>
+```
+
+- With the navigation apps, paramter `audioStreamingState` was extened from the original design purpose (an indicator just for A2DP BT audio service) to include new binary audio service. Along the same line of this proposal, add a new paratmer `isAudioSource` for binary audio service only and make existing `audioStreamingState` just for A2DP aduio. This makes things clear but needs more modifications to the existing code.
```

Comment:
Please update the first sentence to "... parameter `audioStreamingState` was extended from..."
Please update the second sentence to "Along the same lines of this proposal, add a new parameter...just for A2DP audio."

Reply:
changed

Reply:
Please update "extended" and "Along the same lines"

Reply:
fixed